Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleNodeHandle.cpp

```diff
@@ -232,3 +232,10 @@ void WKBundleNodeHandleSetHTMLInputElementAutoFillButtonEnabled(WKBundleNodeHand
 {
     // FIXME: Would put ASSERT_NOT_REACHED() here but some compilers are warning the function is "noreturn".
 }
+
+WKBundleFrameRef WKBundleNodeHandleDocumentFrame(WKBundleNodeHandleRef documentHandleRef)
```

Comment:
Since this returns a +1 object (using `leakRef()`), the caller should use `adoptWK` to wrap it in a smart pointer, to prevent a leak.

We usually denote this behavior by using the name `Copy` in the function itself (like `WKBundleNodeHandleCopyDocumentFrame`), but in this case, that symbol already exists (and behaves slightly differently, requiring the node handle to be a document itself).

WDYT about renaming to something like `WKBundleNodeHandleCopyOwningDocumentFrame`, and also adding `adoptWK` at the call site?